terraform/modules/vpc/endpoints.tf

```diff
@@ -0,0 +1,17 @@
+resource "aws_vpc_endpoint" "concourse" {
+  count             = local.zone_count
+  vpc_id            = module.vpc.vpc.id
+  service_name      = "aws.concourse.endpoint"
+  vpc_endpoint_type = "Interface"
+
+  security_group_ids = [var.loadbalancer.security_group_id]
```

Comment:
I think the endpoint needs to sit in its own SG. This matches the pattern we have for AWS-service related VPC endpoints.

With that in place, you'll need to add SG rules for the following:

Concourse Web egress to VPC endpoint on TCP 3128
VPC endpoint ingress from Concourse Web on TCP 3128
Concourse Worker egress to VPC endpoint on TCP 3128
VPC endpoint ingress from Concourse Worker on TCP 3128